Adding unstuck transaction guide (#111)
sora-xor/sora-docs: src/unstuck-bridge-tx.md

3. Paste in your Approved transaction hash from step 2 instead of 0x000…
4. Click “Include Option” and leave the network id as 0.
5. Submit the RPC Call:

![](/.gitbook/assets/unstucktx-add-approved-request.png)

The result is your “Proof”, it contains all the data needed for step 4.

::: details Expand to see an example of proof

```
Transfer: {

currency_id: {

TokenAddress: 0x6b175474e89094c44da98b954eedeac495271d0f

},

amount: 30,000,000,000,000,000,000,000,

to: 0xd7230339a4b355ae6d24a7378cc910o85l34k5nb,

from: 0x6ef01fd3b42ed7a3fa0e7ca366d0a464dc71016d,

tx_hash: 0x235517c869f0288ad8jjhag6f0de0jbkelf6ooi5ycct5g9d945d73d40d987y6b,

network_id: 0x0000000000000000000000000000000000000000000000000000000000000000,

raw: 0x6b175474e89094c44da98b954eedeac495271d0f00000000000000000000000000000000000000000000069c1d44c523e2f80000d7330339a1b395ae5d24ab376cc610085034458c6
{

r: 0x7a42sefc5dh232j34g61d22c53b28cn9600f4jhggf70fdfeh83kb4l7d9s2bcea,

s: 0x79b1e77bb20663953981d267fbc01d53f9cc3x34s63aa7h3f4385l04td9be9c43,

*v: 0

},

{

r: 0xcc9fcee0fd50bfa29ekb10ldlkjh3tdef02f9beb74a1v295ac0c5x01253f8f8f,

s: 0xkafvd79ffa27g10h928jb88066e8252dec3d58fb6ebf0862c6839d880f858a8f,

v:
```

> :::

Reviewer: ```suggestion
```

# How to Unstuck SORA→ETHEREUM Transactions

::: info
This guide is written by the community member "Frasier Man" and reviewed by the SORA team. If you want thank Fraiser man, please donate to: `5D4pKyFNRjU8yoYKgiH1UqrjzayjCkttt82HsAdwQ6FTsQyv`
:::

> ### Use This Guide If:

Reviewer 2: Why was the second heading skipped?

- Your SORA bridge transaction is stuck on pending or has failed.

> - The "retry" command on the bridge UI is not working/not available.

Reviewer 2: Do you think that we should support this paragraph with the link to Polkaswap or the screenshot of the Retry button?

Author: I can add for more clarity, yes.

- Your funds have left your SORA account and not arrived in your Ethereum account after some time.

:::info

> If at the end of Step 4, Ethereum asks you for a LARGE

Reviewer 2: Would be nice to include the direct links to the steps (that are the headings essentially)

Author: Yes, I was also considering implementing this

transaction fee, DO NOT CONFIRM THE TRANSACTION. It means data has
been entered incorrectly, double check your input.
:::

### Overview of the Steps in this Guide

Step 1. Check if your bridge transaction is actually stuck or is just pending.

Step 2. Find the transaction hash for your stuck transaction.

Step 3. Get the data needed to unstuck the transaction (getting your proof).

Step 4. Use your proof to manually call your Ethereum transaction to receive your tokens.

> **NOTE: Step 4 is slightly different for each token so make sure to follow the correct section in Step 4.**

Reviewer 2: ```suggestion
::: info
Step 4 is slightly different for each token so make sure to follow the correct section in Step 4.
:::
```
```

Click write, this should bring up your metamask for you to confirm. Remember if the gas price is over the top (100s of $) then some of the data is wrong. Once confirmed your VAL should arrive in your wallet, please remember to add the token address to your wallet if necessary.

##### PSWAP & CERES

- Open https://etherscan.io/address/0x1485e9852ac841b52ed44d573036429504f4f602
- Click “Contract”
- Click “Write Contract”
- Click “Connect Web 3” (ensure you have your eth address and a green circle shown after connection)
- Click “receiveBySidechainAssetID”

![](/.gitbook/assets/unstucktx-receive-sidechain-asset.png)

Enter your proof data into the fields as below, this is using our example proof from Step 3, This is how you must paste in your data. (note: sideChainassetid = token address)

![](/.gitbook/assets/unstucktx-sidechain-input.png)

Click write, this should bring up your metamask for you to confirm. Remember if the gas price is over the top (100s of $) then some of the data is wrong. Once confirmed your PSWAP should arrive in your wallet, please remember to add the token address to your wallet if necessary.

##### All Other ERC-20 Tokens (e.g. ETH, DAI, UNI etc.)

> - Open https://etherscan.io/address/0x1485e9852ac841b52ed44d573036429504f4f602
> - Click “Contract”
> - Click “Write Contract”
> - Click “Connect Web 3” (ensure you have your eth address and a green circle after connection)
> - Click “receiveByEthereumAssetAddress”
> 
> ![](/.gitbook/assets/unstucktx-eth-asset-address.png)
> 
> Enter your proof data into the fields as below, this is using our example proof from Step 3, This is how you must paste in your data.
> 
> ![](/.gitbook/assets/unstucktx-eth-asset-proof.png)
> 
> Click write, this should bring up your metamask for you to confirm.
> **Remember if the gas price is over the top (100s of $) then some of
> the data is wrong.**
> Once confirmed your tokens should arrive in your wallet, please remember to add the token address to your wallet if necessary.

Reviewer: ```suggestion
1. Open https://etherscan.io/address/0x1485e9852ac841b52ed44d573036429504f4f602
2. Click “Contract”
3. Click “Write Contract”
4. Click “Connect Web 3”
5. Ensure you have your eth address and a green circle after connection
6.  Click “receiveByEthereumAssetAddress”

![](/.gitbook/assets/unstucktx-eth-asset-address.png)

7. Enter your proof data into the fields as below, this is using our example proof from Step 3. This is how you must paste in your data.

![](/.gitbook/assets/unstucktx-eth-asset-proof.png)

8. Click "write", this should bring up your metamask for you to confirm.
**Remember if the gas price is over the top (100s of $) then some of
the data is wrong.**
Once confirmed your tokens should arrive in your wallet. Please remember to add the token address to your wallet if necessary.
```

---
title: "How to Unstuck SORA-ETHEREUM Transactions | Troubleshooting Guide"
head:
  - - meta
    - name: description
      content: "Learn how to resolve stuck transactions between the SORA Network and Ethereum. Follow this step-by-step guide to resolve stuck or failed SORA-ETHEREUM transactions. Learn how to check the status of your bridge transaction, retrieve the transaction hash, gather the necessary data to unstuck the transaction, and manually call your Ethereum transaction to receive your tokens."
  - - meta
    - name: keywords
      content: "SORA-ETHEREUM transactions, Sora>ETH bridge, Ethereum transaction, unstuck transactions, metamask, cross-chain transfers, troubleshooting, SORA Network, Ethereum, bridge transaction status, transaction hash, unstuck transaction, troubleshooting guide"
---

# How to Unstuck SORA→ETHEREUM Transactions

::: info

> This guide is written by the community member "Frasier Man" and reviewed by the SORA team. If you want thank Fraiser man, please donate to: `5D4pKyFNRjU8yoYKgiH1UqrjzayjCkttt82HsAdwQ6FTsQyv`

Reviewer: ```suggestion
This guide is written by the community member "Frasier Man" and reviewed by the SORA team. If you want to thank Fraiser man, please donate to: `5D4pKyFNRjU8yoYKgiH1UqrjzayjCkttt82HsAdwQ6FTsQyv`
```

:::

> ### Use This Guide If:

Reviewer: I think "When to use this guide" would be a better title

Author: Changed

- Your SORA bridge transaction is stuck on pending or has failed.
- The "retry" command on the [bridge UI](https://polkaswap.io/#/bridge/) is not working/not available.
- Your funds have left your SORA account and not arrived in your Ethereum account after some time.

:::info
If at the end of Step 4, Ethereum asks you for a LARGE
transaction fee, DO NOT CONFIRM THE TRANSACTION. It means data has
been entered incorrectly, double check your input.
:::

> 

Reviewer: I'd suggest to
1)  move this note in the overview section
2)  add a similar one in step 4 where it's more relevant

## Overview of the Steps in this Guide

- [Step 1](#Step-1) Check if your bridge transaction is actually stuck or is just pending.

- [Step 2](#step-2) Find the transaction hash for your stuck transaction.

- [Step 3](#step-3) Get the data needed to unstuck the transaction (getting your proof).

- [Step 4](#step-4) Use your proof to manually call your Ethereum transaction to receive your tokens.

> **NOTE: [Step 4](#step-4) is slightly different for each token so make sure to
> follow the correct subsection.**

Reviewer: ```suggestion
::: info
[Step 4](#step-4) is slightly different for each token so make sure to
follow the correct subsection.**
:::
```

### Step 1

Check if your transaction is actually stuck or just pending.

> - Connect to Polkadot.js apps: [https://polkadot.js.org/apps/rpc=wss%3A%2F%2Fws.sora2.soramitsu.co.jp#/rpc](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fws.sora2.soramitsu.co.jp#/rpc)
> 
> - Click "Developer", then Select "RPC calls"
>   ![](/.gitbook/assets/unstucktx-rpc-call.png)
> 
> - Click “author”, then Select “Ethbridge” (by default “getAccountRequests(accountId, statusFilter)” should be selected, if not, select it.
>   ![](/.gitbook/assets/unstucktx-ethbridge.png)
> 
> - Ensure your account is selected underneath, click on "include option" and choose "Pending", then Submit RPC call
>   ![](/.gitbook/assets/unstucktx-include-option.png)

Reviewer: ```suggestion
1. Connect to Polkadot.js app: [https://polkadot.js.org/apps/rpc=wss%3A%2F%2Fws.sora2.soramitsu.co.jp#/rpc](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fws.sora2.soramitsu.co.jp#/rpc)

2. Click "Developer", then Select "RPC calls":
  ![](/.gitbook/assets/unstucktx-rpc-call.png)

3. Click “author”, then Select “Ethbridge”. By default “getAccountRequests(accountId, statusFilter)” should be selected, if not, select it.
  ![](/.gitbook/assets/unstucktx-ethbridge.png)

4. Ensure your account is selected underneath, click on "include option" and choose "Pending", then Submit RPC call
  ![](/.gitbook/assets/unstucktx-include-option.png)
```

> If your result contains any transaction hashes, these ARE still pending and may take some time to go through. **If your result looks as below and produces no hashes then your SORA bridge transaction has been approved and you should continue through the guide to unstuck your transaction.**

Reviewer: ```suggestion
If your result contains any transaction hashes, these ARE still pending and may take some time to go through. **If your result looks as below and produces no hashes, then your SORA bridge transaction has been approved and you should continue through the guide to unstuck your transaction.**
```

![](/.gitbook/assets/unstucktx-approved-continue.png)

### Step 2

Find the transaction hash of your approved (but stuck) transaction.

- If continuing from step 1 your RPC call only needs slight adjusting, click on “include option” but this time select “Approvals Ready”, Then Submit RPC Call
  ![](/.gitbook/assets/unstucktx-approvals-ready.png)

The result is a list of all SORA bridge transactions that have been
approved on the SORA side. Your latest bridge transaction is the last
one (scroll down if necessary), if you have made more SORA>ETH bridge
transactions after the stuck one, you will need to go from the bottom
to the top of the list until you find your stuck transaction.

> - Copy your “stuck” transaction hash, it will look like this:

Reviewer: ```suggestion
1. If continuing from step 1, your RPC call only needs slight adjusting. Click on “include option” but this time select “Approvals Ready”, then Submit RPC Call
  ![](/.gitbook/assets/unstucktx-approvals-ready.png)

The result is a list of all SORA bridge transactions that have been
approved on the SORA side. Your latest bridge transaction is the last
one (scroll down if necessary). If you have made more SORA>ETH bridge
transactions after the stuck one, you will need to go from the bottom
to the top of the list until you find your stuck transaction.

2. Copy your “stuck” transaction hash, it will look like this:
```

`0x57533dc46a5d257b6dj0nhagb0f8s4dfedfl6cpppcoaidu99dd0sdc4def4g6b`

![](/.gitbook/assets/unstucktx-stuck-tx.png)

### Step 3

Use your transaction hash to get the required data (proof) for step 4.

> - Continuing from step 2, click on “get**Account**Requests(accountId, statusFilter)” and change it to “get**Approved**Requests(requestHashes, networkId)
> 
> ![](/.gitbook/assets/unstucktx-get-approved-requests.png)
> 
> - Click on “Add item”
> - Paste in your Approved transaction hash from step 2 instead of 0x000…
> - Click “Include Option” and leave the network id as 0
> - Submit the RPC Call
> 
> ![](/.gitbook/assets/unstucktx-add-approved-request.png)

Reviewer: ```suggestion
1. Continuing from step 2, click on “get**Account**Requests(accountId, statusFilter)” and change it to “get**Approved**Requests(requestHashes, networkId)

![](/.gitbook/assets/unstucktx-get-approved-requests.png)

2. Click on “Add item”.
3. Paste in your Approved transaction hash from step 2 instead of 0x000…
4. Click “Include Option” and leave the network id as 0.
5. Submit the RPC Call:

![](/.gitbook/assets/unstucktx-add-approved-request.png)

```

The result is your “Proof”, it contains all the data needed for step
4, below is an **example** of a proof, you only need the data colour
highlighted in this example, the rest can be ignored, underneath this
example are instructions on how yo should prepare **your** data for the next step.

> 

Reviewer: ```suggestion
The result is your “Proof”, it contains all the data needed for step 4.

::: details Expand to see an example of proof
```

```
Transfer: {

currency_id: {

TokenAddress: 0x6b175474e89094c44da98b954eedeac495271d0f

},

amount: 30,000,000,000,000,000,000,000,

to: 0xd7230339a4b355ae6d24a7378cc910o85l34k5nb,

from: 0x6ef01fd3b42ed7a3fa0e7ca366d0a464dc71016d,

tx_hash: 0x235517c869f0288ad8jjhag6f0de0jbkelf6ooi5ycct5g9d945d73d40d987y6b,

network_id: 0x0000000000000000000000000000000000000000000000000000000000000000,

raw: 0x6b175474e89094c44da98b954eedeac495271d0f00000000000000000000000000000000000000000000069c1d44c523e2f80000d7330339a1b395ae5d24ab376cc610085034458c6
{

r: 0x7a42sefc5dh232j34g61d22c53b28cn9600f4jhggf70fdfeh83kb4l7d9s2bcea,

s: 0x79b1e77bb20663953981d267fbc01d53f9cc3x34s63aa7h3f4385l04td9be9c43,

*v: 0

},

{

r: 0xcc9fcee0fd50bfa29ekb10ldlkjh3tdef02f9beb74a1v295ac0c5x01253f8f8f,

s: 0xkafvd79ffa27g10h928jb88066e8252dec3d58fb6ebf0862c6839d880f858a8f,

v:
```

> 

Reviewer: ```suggestion
:::

```

> **You will be copying and pasting data from your proof. The first time you do this, it's easier if you first copy and paste the whole proof into a notepad, and edit the results as follows, ready to use into step 4.**

Reviewer: ```suggestion
Here's the guide on how yuo should prepare **your** data for the next step.

::: tip
You will be copying and pasting data from your proof. The first time you do this, it's easier if you first copy and paste the whole proof into a notepad, and edit the results as described below, ready to use into step 4.
:::
```

> - Token Address

Reviewer: this part better works as a table, e.g. something like this:

|Field|Guidance|Example|
|--|--|--|
|Token Address|This can be copied and pasted as it is, no commas or spaces. | `0x6b175474e89094c44da98b954eedeac495271d0f`|
|Amount|Copy and paste this number and delete all the commas. This amount will reflect the amount you bridged but will be a much larger number, for example the amount in this example bridge was actually 30000. | `30000000000000000000000`|
|...|...|.....|

or maybe a set of multiple tables since for r, s, v fields we need to add an additional note beforehand. Alternatively, you can have the basic info about r, s, v as part of the bigger table but then add a section specifically about these values since there are a lot of details

  tokens. **For all tokens except XOR**

If your `v = 0` then you must put `27` into the array

If your`v = 1` then you must put `28` into the array

For our example, top v is`0` and bottom v is `1`so our array would be

`[27,28]`

- For XOR

If `v = 0` then you must put `0x1b` into the array, If `v = 1` then you must put `0x1c` into the array

For our example, top v is `0` and bottom v is `1` so our array would be

`[0x1b,0x1c]`

- Result from Proof

You now have everything you need for step 4, from our example proof we have the following, yours will look similar to this.

```
0x6b175474e89094c44da98b954eedeac495271d0f

30000000000000000000000

0xd7230339a4b355ae6d24a7378cc910o85l34k5nb

0x6ef01fd3b42ed7a3fa0e7ca366d0a464dc71016d

0x235517c869f0288ad8jjhag6f0de0jbkelf6ooi5ycct5g9d945d73d40d987y6b

[0x7a42sefc5dh232j34g61d22c53b28cn9600f4jhggf70fdfeh83kb4l7d9s2bcea,0xcc9fcee0fd50bfa29ekb10ldlkjh3tdef02f9beb74a1v295ac0c5x01253f8f8f]

[0x79b1e77bb20663953981d267fbc01d53f9cc3x34s63aa7h3f4385l04td9be9c43,0xkafvd79ffa27g10h928jb88066e8252dec3d58fb6ebf0862c6839d880f858a8f]

[27,28] (for all tokens except XOR) or [0x1b,0x1c] (for XOR)
```

### Step 4

> Using the data from your proof to call the ethereum transaction

Reviewer: Use the data ... to call ...
Using the data..., call ...

This sentence is unfinished and is not clear

Author: This is an overview of what step 4 is, all the steps above have the same sentence summarising the step.

Reviewer: it's not the same though

```suggestion
Use the data from your proof to call the ethereum transaction.
```

There are four different methods depending on the token you are
unstucking, skip to the correct method and have your proof data ready
to paste.

- [XOR](#xor)
- [VAL](#val)
- [PSWAP & CERES](#pswap-ceres)
- [All other ERC-20 tokens](#all-other-erc-20-tokens-e-g-eth-dai-uni-etc)

##### XOR

> - Open https://web3playground.io/QmfPMVGCCSHYtk5Z6DYYzv5hZ1SDakpB3bwA5w9rDTPB9A
> - Your metamask will prompt you to connect, please ensure the correct wallet is selected.
> - Input your proof data into the top fields within the code. NOTE: for XOR you must only paste in the data itself, leave out the array brackets and commas as they are already in place in the code.
> - There is an example proof already on the page, you must only replace the data, leaving the quotation marks in place.
> 
> Below is the sample data you will see when opening the link, please note you may have to scroll right to see the whole of the code, the green words (e.g. // Token address) are to let you know which data to paste in.
> 
> ![](/.gitbook/assets/unstucktx-link-sample-data.png)
> 
> Below is our example proof from STEP 3 entered into the code, note how the quote characters must be left in place. This is how you must paste in your data.
> 
> ![](/.gitbook/assets/unstucktx-paste-data-example.png)
> 
> - Click run, this should bring up your metamask for you to confirm. Remember if the gas price is over the top (100s of $) then some of the data is wrong. Once confirmed your XOR should arrive in your wallet, please remember to add the token address to your wallet if necessary.

Reviewer: ```suggestion
1. Open https://web3playground.io/QmfPMVGCCSHYtk5Z6DYYzv5hZ1SDakpB3bwA5w9rDTPB9A. 
2. Your metamask will prompt you to connect, please ensure the correct wallet is selected.
3. Enter your proof data into the top fields within the code. Note that for XOR you must only paste in the data itself, leave out the array brackets and commas as they are already in place in the code.
4. There is an example proof already on the page, you must only replace the data, leaving the quotation marks in place.

Below is the sample data you will see when opening the link, please note you may have to scroll right to see the whole of the code, the green words (e.g. // Token address) are to let you know which data to paste in.

![](/.gitbook/assets/unstucktx-link-sample-data.png)

Below is our example proof from STEP 3 entered into the code, note how the quote characters must be left in place. This is how you must paste in your data.

![](/.gitbook/assets/unstucktx-paste-data-example.png)

5. Click run, this should bring up your metamask for you to confirm. Remember if the gas price is over the top (hundreds of dollars), then some of the data is wrong. Once confirmed your XOR should arrive in your wallet. Please remember to add the token address to your wallet if necessary.
```

##### VAL

> - Open https://etherscan.io/address/0xd1eeb2f30016fffd746233ee12c486e7ca8efef1
> - Click “Contract”
> - Click “Write as Proxy”
> - Click “Connect Web 3” (ensure you have your eth address and a green circle shown after connection)
> - Click “mintTokensByPeers”
> 
> ![](/.gitbook/assets/unstucktx-mint-tokens-peers.png)
> 
> Enter your proof data into the fields as below, this is using our example proof from Step 3, This is how you must paste in your data. (note: beneficiary = to)
> 
> ![](/.gitbook/assets/unstucktx-proof-input.png)
> 
> Click write, this should bring up your metamask for you to confirm. Remember if the gas price is over the top (100s of $) then some of the data is wrong. Once confirmed your VAL should arrive in your wallet, please remember to add the token address to your wallet if necessary.

Reviewer: ```suggestion
1. Open https://etherscan.io/address/0xd1eeb2f30016fffd746233ee12c486e7ca8efef1
2. Click “Contract”
3. Click “Write as Proxy”
4. Click “Connect Web 3”. 
5. Ensure you have your eth address and a green circle shown after connection.
6. Click “mintTokensByPeers”

![](/.gitbook/assets/unstucktx-mint-tokens-peers.png)

7. Enter your proof data into the fields as below, this is using our example proof from Step 3, This is how you must paste in your data. (note: beneficiary = to)

![](/.gitbook/assets/unstucktx-proof-input.png)

8. Click "write", this should bring up your metamask for you to confirm. Remember if the gas price is over the top (hundreds of dollars), then some of the data is wrong. Once confirmed your VAL should arrive in your wallet. Please remember to add the token address to your wallet if necessary.
```

##### PSWAP & CERES

> - Open https://etherscan.io/address/0x1485e9852ac841b52ed44d573036429504f4f602
> - Click “Contract”
> - Click “Write Contract”
> - Click “Connect Web 3” (ensure you have your eth address and a green circle shown after connection)
> - Click “receiveBySidechainAssetID”
> 
> ![](/.gitbook/assets/unstucktx-receive-sidechain-asset.png)
> 
> Enter your proof data into the fields as below, this is using our example proof from Step 3, This is how you must paste in your data. (note: sideChainassetid = token address)
> 
> ![](/.gitbook/assets/unstucktx-sidechain-input.png)
> 
> Click write, this should bring up your metamask for you to confirm. Remember if the gas price is over the top (100s of $) then some of the data is wrong. Once confirmed your PSWAP should arrive in your wallet, please remember to add the token address to your wallet if necessary.

Reviewer: ```suggestion
1. Open https://etherscan.io/address/0x1485e9852ac841b52ed44d573036429504f4f602
2. Click “Contract”
3. Click “Write Contract”
4. Click “Connect Web 3”
5. Ensure you have your eth address and a green circle shown after connection
6. Click “receiveBySidechainAssetID”

![](/.gitbook/assets/unstucktx-receive-sidechain-asset.png)

7. Enter your proof data into the fields as below, this is using our example proof from Step 3. This is how you must paste in your data. (note: sideChainassetid = token address)

![](/.gitbook/assets/unstucktx-sidechain-input.png)

8. Click "write", this should bring up your metamask for you to confirm. Remember if the gas price is over the top (hundreds of dollars), then some of the data is wrong. Once confirmed your PSWAP should arrive in your wallet. Please remember to add the token address to your wallet if necessary.
```

##### All Other ERC-20 Tokens (e.g. ETH, DAI, UNI etc.)

> - Open https://etherscan.io/address/0x1485e9852ac841b52ed44d573036429504f4f602
> - Click “Contract”
> - Click “Write Contract”
> - Click “Connect Web 3” (ensure you have your eth address and a green circle after connection)
> - Click “receiveByEthereumAssetAddress”
> 
> ![](/.gitbook/assets/unstucktx-eth-asset-address.png)
> 
> Enter your proof data into the fields as below, this is using our example proof from Step 3, This is how you must paste in your data.
> 
> ![](/.gitbook/assets/unstucktx-eth-asset-proof.png)
> 
> Click write, this should bring up your metamask for you to confirm. Remember if the gas price is over the top (100s of $) then some of the data is wrong. Once confirmed your tokens should arrive in your wallet, please remember to add the token address to your wallet if necessary.

Reviewer: ```suggestion
1. Open https://etherscan.io/address/0x1485e9852ac841b52ed44d573036429504f4f602
2. Click “Contract”
3. Click “Write Contract”
4. Click “Connect Web 3”
5. Ensure you have your eth address and a green circle after connection
6. Click “receiveByEthereumAssetAddress”

![](/.gitbook/assets/unstucktx-eth-asset-address.png)

7. Enter your proof data into the fields as below, this is using our example proof from Step 3. This is how you must paste in your data.

![](/.gitbook/assets/unstucktx-eth-asset-proof.png)

8. Click "write", this should bring up your metamask for you to confirm. Remember if the gas price is over the top (hundreds of dollars), then some of the data is wrong. Once confirmed your tokens should arrive in your wallet. Please remember to add the token address to your wallet if necessary.
```
